core/domain/skill_jobs_one_off_test.py

```diff
@@ -169,3 +169,45 @@ def test_migration_job_converts_old_skill(self):
         expected = [[u'skill_migrated',
                      [u'1 skills successfully migrated.']]]
         self.assertEqual(expected, [ast.literal_eval(x) for x in output])
+
+    def test_migration_job_skips_when_new_structure_editor_is_disabled(self):
+        skill = skill_domain.Skill.create_default_skill(
+            self.SKILL_ID, description='A description')
+        skill_services.save_new_skill(self.albert_id, skill)
+
+        with self.swap(constants, 'ENABLE_NEW_STRUCTURE_EDITORS', False):
+            job_id = (
+                skill_jobs_one_off.SkillMigrationOneOffJob.create_new())
+            skill_jobs_one_off.SkillMigrationOneOffJob.enqueue(job_id)
+            self.process_and_flush_pending_tasks()
+
+            output = skill_jobs_one_off.SkillMigrationOneOffJob.get_output(
+                job_id)
+            self.assertEqual(output, [])
+
+    def test_migration_job_skips_updated_skill_failing_validation(self):
+
+        def _mock_get_skill_by_id(unused_skill_id):
+            """Mocks get_skill_by_id()."""
+            return 'invalid_skill_id'
```

Comment:
Semantically, this should be 'invalid_skill', right? Since this method is supposed to return a skill.

Ditto in other files.

Reply:
Done